contrib/devtools/xversionkeys.py

```diff
@@ -97,6 +100,20 @@ def writeTableCPPH(outfile, table):
         print ("    {%40s, %42s }," % (x["name"], "xvt_"+x["valtype"]), file = outfile)
     print("}; // const unordered_map valtype\n\n\n", file = outfile)
 
+    # set key types in enum
+    print("enum {", file = outfile)
+    valtypes = set(x["keytype"] for x in table.values())
+    for x in sorted(valtypes):
+        print ("    %20s," % ("xkt_"+x), file = outfile)
+    print("}; // enum keytypes\n\n\n", file = outfile)
```

Comment:
I would go and always make the enum complete - extract from the `keytypes` variable you defined above instead of the parsed `table`. This also will ensure you can actually refer to these values, no matter the actual table, in the more generic parts of the code.

Reply:
this is a good point. i will adjust that